pulp_docker/app/viewsets.py

```diff
@@ -172,3 +174,75 @@ class DockerDistributionViewSet(BaseDistributionViewSet):
     endpoint_name = 'docker'
     queryset = models.DockerDistribution.objects.all()
     serializer_class = serializers.DockerDistributionSerializer
+
+
+class TagImageViewSet(viewsets.ViewSet):
+    """
+    ViewSet used for tagging manifests. This endpoint supports only HTTP POST requests.
+    """
+
+    endpoint_name = 'tag'
+
+    @swagger_auto_schema(
+        operation_description="Trigger an asynchronous task to create a new repository",
+        responses={202: AsyncOperationResponseSerializer}
+    )
+    def create(self, request):
+        """
+        Create a task which is responsible for initializing a new repository version.
```

Comment:
I consider a new repository version to be almost a side effect. The action of this task is that it tags a manifest.